ydb/library/yql/parser/pg_wrapper/arrow.cpp

```diff
@@ -151,6 +152,84 @@ std::shared_ptr<arrow::Array> PgConvertString(const std::shared_ptr<arrow::Array
     return ret;
 }
 
+Numeric PgFloatToNumeric(double item, ui64 scale, int digits) {
+    double intPart, fracPart;
+    bool error;
+    fracPart = modf(item, &intPart);
+    i64 fracInt = round(fracPart * scale);
+
+    // scale compaction: represent 711.56000 as 711.56
+    while (fracInt && fracInt % 10 == 0) {
+        fracInt /= 10;
+        digits -= 1;
+    }
+
+    return numeric_add_opt_error(
+        int64_to_numeric(intPart),
+        int64_div_fast_to_numeric(fracInt, digits),
+        &error);
+}
+
+template<typename T>
+std::shared_ptr<arrow::Array> PgConvertNumeric(const std::shared_ptr<arrow::Array>& value) {
+    TArenaMemoryContext arena;
+    const auto& data = value->data();
+    size_t length = data->length;
+    arrow::BinaryBuilder builder;
```

Comment:
please use StringBuilder from public/udf/arrow

Reply:
I spoke with @nepal. Now this cannot be done, we need to do a small refactoring for which there will be a separate ticket.